### ISSUE TITLE
Model loading via SQLAlchemy

### ISSUE THREAD

**Issue by mactyr** [energyPATHWAYS/energyPATHWAYS#40]
[technical ticket, no story]

This ticket tracks the development of a first implementation of the Pathways "model layer", based on SQLAlchemy.

**AC**
1. All core Pathways objects are loaded through SQLAlchemy. In most cases, this will mean that existing classes tied to database tables are converted to subclasses of a SQLAlchemy declarative_base. Some data structures that don't fit neatly into the single-table (or single-table + data table) model object paradigm (e.g. the geography map) may be loaded more directly via SQLAlchemy's query capabilities. We will know this ticket is complete when:
   1. We have deleted all the low-level database access functions in util and config
   2. We have removed all direct references to a database driver (e.g. psycopg2) in the code
2. Unit tests give us a high degree of confidence that the model loaded is identical to the one loaded before the transition to SQLAlchemy. For now, these unit tests are permitted to be dependent on the us_model_example database.

**Out of Scope / Future Work**
- Validation of model objects
- RESTful/json API for instantiating and validating models from the web interface
- Database-independent unit testing of model classes
- Explicit work on performance improvements (I expect that loading tables in bulk through SQLAlchemy rather than one-at-a-time will get us a significant reduction in model load time "for free" but I don't intend to push any further in that direction than just using basic SQLAlchemy capabilities).

**Dependencies**

Depends on #39.

**Comment by mactyr** [comment]
We have decided to handle this incrementally by "clusters" of tables (e.g. system tables, demand subsector related tables, demand tech related tables, etc.) and do some database consolidation at the same time. Rather than gut and re-write this ticket, which I think might be confusing, I will close it as invalid and begin a new series of tickets to reflect the new task organization (starting with #42).
